File: contains_duplicate/contains_duplicate.ts
// My Answer
function containsDuplicate(nums: number[]): boolean {
	nums.sort();
	for(let i = 0; i < nums.length; i++) {
		if(nums[i] == nums[i+1]) {
			return true;
		}
	}
	return false;
};

// Using Set
function containsDuplicateTwo(nums: number[]): boolean {
	const set = new Set();
	for (const num of nums) {
		if(set.has(num)) {
			return true;
		}
		set.add(num);
	}
	return false;
}

File: contains_duplicate/contains_duplicate.java
package contains_duplicate;

import java.util.ArrayList;
import java.util.Arrays;

class MyAnswer {
	public boolean containsDuplicate(int[] nums) {
		var seenNumbers = new ArrayList<Integer>();
		for (int num : nums) {
			if (seenNumbers.contains(num)) {
				return true;
			} else {
				seenNumbers.add(num);
			}
		}
		return false;
	}
}

class UsingSort {
	public boolean containsDuplicate(int[] nums) {
		Arrays.sort(nums);
		for(int i = 0; i < nums.length; i++) {
			if(nums[i] == nums[i+1]) {
				return true;
			}
		}
		return false;
	}
}  